### PR TITLE
NAS-110413 / 21.06 / Fix form issues in data-protection dashboard

### PR DESCRIPTION
**Pull request by dszidi**
Scrub/Snapshot task forms now update form values from correct row.

To test the snapshot or scrub task forms:
- Make sure there are multiple rows in the table
- Ensure that the edit form for each row actually populates all the fields with the correct row data.
- Also make sure that custom select option in form-scheduler works as expected.

### CODE REVIEW

File: src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts [hunk 1]
```diff
@@ -223,4 +225,22 @@ export class SnapshotFormComponent implements FormConfiguration, OnDestroy {
...
   }
+
+  dataHandler(entity: EntityFormComponent): void {
+    // Setup cron_schedule
+    const schedule = entity.wsResponse.schedule;
+    if (Number(entity.wsResponse.id) !== Number(this.pk)) console.error({ id: entity.wsResponse.id, pk: this.pk });
+    const formatted = schedule.minute + ' ' + schedule.hour + ' ' + schedule.dom + ' ' + schedule.month + ' ' + schedule.dow;
+    const cronField = entity.formGroup.controls['cron_schedule'];
+    cronField.setValue(formatted);
+    entity.fieldConfig[5].value = formatted;
```
Comment (denysbutenko) [inline src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts:236]: index usage looks not very reliable. can it be more specific?

File: src/app/pages/common/entity/entity-form/components/form-scheduler/form-scheduler.component.ts [hunk 2]
```diff
@@ -259,7 +278,7 @@ export class FormSchedulerComponent implements Field, OnInit, AfterViewInit, Aft
     protected localeService: LocaleService, protected ws: WebSocketService,
     private sysGeneralService: SystemGeneralService) {
     // Set default value
-    this.preset = this.presets[1];
+    this.preset; // = this.presets[1];
```
Comment (denysbutenko) [inline src/app/pages/common/entity/entity-form/components/form-scheduler/form-scheduler.component.ts:281]: seems it can be removed

**Review by denysbutenko — APPROVED** [review]
Works as described!
image

File: src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts (hunk 1 above)
Comment (dszidi) [reply src/app/pages/data-protection/snapshot/snapshot-form/snapshot-form.component.ts]: Yep. I did update that for scrub form but apparently missed it on the snapshot form.  Nice catch.

**Pull request merged**